app/controllers/merchants_controller.rb

```diff
@@ -0,0 +1,40 @@
+class MerchantsController < ApplicationController
+  def index
+    @merchants = Merchant.all
+  end
+
+  def new
+    @merchant = Merchant.new
+  end
+
+  def show
+    @merchant = Merchant.find(params[:id])
+  end
+
+  def create
+    merchant = Merchant.create!(merchant_params)
+    merchant.save
+    redirect_to "/merchants"
+  end
+
+  def edit
+    @merchant = Merchant.find(params[:id])
+  end
```

Comment:
This is good.. since when you go to the edit view (which has the form to edit an item) you need to know which merchant you're editing when filling out and submitting that edit form. This is why it's necessary here to assign `@merchant1` so that you have it to use in your view (in your case, it would be used so that we can then pass it to the update method when trying to update this merchant upon clicking submit)